ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/InMemoryKVParticipantState.scala

```diff
@@ -125,6 +139,62 @@ class InMemoryKVParticipantState(implicit system: ActorSystem, mat: Materializer
       )
     )
 
+  /** Akka actor that matches the requests for party allocation
+    * with asynchronous responses delivered within the log entries.
+    */
+  class ResponseMatcher extends Actor {
```

Comment:
I'd be interested in seeing how the Sawtooth implementation of this logic ends up looking like and whether we can abstract it out and use it for other implementations as well.